packages/app-storage/src/Query.tsx

```diff
@@ -41,11 +42,48 @@ class Query extends React.PureComponent<Props, State> {
         value as Storage$Key$Value
       );
 
+      const renderAccounts = (accountIds: Array<string>) => {
+        return (
+          <table className='accounts'>
+            <thead>
+              <tr>
+                <th />
+                <th>Account ID</th>
+              </tr>
+            </thead>
+            <tbody>
+              {
+                accountIds && accountIds.length ?
+                  (
+                    accountIds
+                      .map((accountId) => (
+                        <tr key={accountId}>
+                          <td><IdentityIcon size={24} value={accountId} /></td>
+                          <td>{accountId}</td>
+                        </tr>
+                      ))
+                  )
+                : null
+              }
+            </tbody>
+          </table>
+        );
+      };
+
+      const lineHeight = key.type[0] === 'AccountId' ? '2em' : '1em';
+
       cache[id] = withStorageDiv(key, { params: values })(
-        (value: any) =>
-          valueToText(key.type, value),
-        { className: 'ui--output' }
+        (value: any) => {
+          const accountIds: Array<string> = valueToText(key.type, value).split(', ');
+          if (key.type[0] === 'AccountId') {
```

Comment:
This should not be in here - the generic valueToText should just render this. There should be no specific rendering in here to do with types at all.

The above check would only work in some specific instances.

Reply:
Basically -

1. valueToText should now return a React.Node
2. with AccountId rendering, pass pack a div/span with the icon & value

This https://github.com/polkadot-js/apps/blob/master/packages/ui-app/src/Params/valueToText.ts#L64 should be the only place where we actually check for the type, nowhere elase.